client/app/models/scheduleitem.coffee

```diff
@@ -309,18 +309,21 @@ module.exports = class ScheduleItem extends Backbone.Model
         # attributes in the model.
         frozenModel = model.clone()
 
-        @confirmSendEmails method, (sendMails) ->
-            # overrides the url to append the sendmails parameter
-            options.url = "#{model.url()}?sendMails=#{sendMails}"
-            return super method, frozenModel, options
+        @confirmSendEmails method, (sendMails) =>
+            @hasShare method, (share) ->
+                # overrides the url to append the sendmails parameter
+                options.url  = "#{model.url()}?sendMails=#{sendMails}"
+                options.url += "&share=#{share}"
+                return super method, frozenModel, options
+
 
     confirmSendEmails: (method, callback) ->
         if @get 'import' # No mails on files import.
             return callback false
 
         # Kind of changes which doesn't need mails.
         if method in ['update', 'patch'] and
-           not ( @startDateChanged or @attendeesChanged)
```

Comment:
I find it confusing to have the second part of the condition at the same indentation level as the return statement, this is why I removed it. I can add another indent if you prefer.